alf/networks/pinverse_network.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (c) 2020 Horizon Robotics and ALF Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""PinverseNetwork"""
+
+import gin
+import functools
+import math
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import alf.utils.math_ops as math_ops
+import alf.nest as nest
+from alf.initializers import variance_scaling_init
+from alf.tensor_specs import TensorSpec
+
+from .network import Network
+from .encoding_networks import EncodingNetwork
+from .projection_networks import NormalProjectionNetwork
+
+
+@gin.configurable
+class PinverseNetwork(Network):
```

Comment:
The name "PinverseNetwork" is not very descriptive.  And need to include more detail on what exactly this network does.